.github/workflows/reusable_test_workflow.yml

```
@@ -0,0 +1,56 @@
+name: Reusable Test Workflow
+on:
+  workflow_call:
+    inputs:
+      julia-version:
+        description: "The Julia version that is used for testing"
+        required: true
+        type: string
+jobs:
+  test:
+    permissions:
+      actions: write # Required for deleting old caches: https://github.com/julia-actions/cache#usage
+      contents: read # Required for checking out the repository
+      id-token: write # Required for interacting with GitHub's OIDC Token endpoint:
+      # For more info, see the OIDC docs:
+      # https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#adding-permissions-settings
```

As workflows get more complex eventually permissions sets need to be merged. Due to that I usually document the required permissions above and not inline as things can get messy fast: 
```suggestion
    # These permissions are needed to:
    # - Checking out the repository (`contents: read`)
    # - Interact with GitHub's OIDC Token endpoint: https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services#adding-permissions-settings
    # - Delete old caches: https://github.com/julia-actions/cache#usage
    permissions:
      actions: write
      contents: read
      id-token: write
```